### PR TITLE
Ncsd 2310 build and publish container

### PR DESCRIPTION
**Pull request by NickGraham101**

### CODE REVIEW

File: Resources/AzureDevOps/StepTemplates/BuildAndPublishContainer.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  DockerfileName: ''
+  ImageName: ''
+
+steps:
+- script: |
+    cd ./Resources/Dockerfiles
+    ImageName=${{ parameters.ImageName }}
+    # Lowercase the AppImage name to conform to Docker Image naming constraints
+    ImageName=${ImageName,,}
+    docker build -t ${{ parameters.ContainerRegistryAdminUser }}.azurecr.io/$ImageName --file ${{ parameters.DockerfileName }}
```
Comment (thelious1974) [inline Resources/AzureDevOps/StepTemplates/BuildAndPublishContainer.yml]: This parameter isn't defined above

File: Resources/AzureDevOps/StepTemplates/BuildAndPublishContainer.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  displayName: 'Build ${{ parameters.ImageName }} container'
+- script: |
+    docker login -u ${{ parameters.ContainerRegistryAdminUser }} -p ${{ parameters.ContainerRegistryPassword }} ${{ parameters.ContainerRegistryAdminUser }}.azurecr.io
```
Comment (thelious1974) [inline Resources/AzureDevOps/StepTemplates/BuildAndPublishContainer.yml]: Same here for 'ContainerRegistryPassword'

File: Resources/AzureDevOps/JobTemplates/BuildAndPublishLinuxContainers.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+parameters:
```
Comment (thelious1974) [inline Resources/AzureDevOps/JobTemplates/BuildAndPublishLinuxContainers.yml]: Suspect these need to be pushed thru to the 'BuildAndPublishContainers' step and the parameters added there too

File: Resources/AzureDevOps/StepTemplates/BuildAndPublishContainer.yml (hunk 2 above)
Comment (NickGraham101) [reply Resources/AzureDevOps/StepTemplates/BuildAndPublishContainer.yml]: Added param

**Pull request merged**